pkg/build/nodeimage/build.go

```diff
@@ -46,21 +49,49 @@ func Build(options ...Option) error {
 		ctx.logger.Warnf("unsupported architecture %q", ctx.arch)
 	}
 
-	// locate sources if no kubernetes source was specified
-	if ctx.kubeRoot == "" {
-		kubeRoot, err := kube.FindSource()
-		if err != nil {
-			return errors.Wrap(err, "error finding kuberoot")
+	if ctx.kubeParam != "" {
+		kubever, err := version.ParseSemantic(ctx.kubeParam)
+		if err == nil {
+			builder, err := kube.NewRemoteBuilder(ctx.logger, "v"+kubever.String(), ctx.arch)
+			if err != nil {
+				return err
+			}
+			ctx.builder = builder
+		} else {
+			if _, err := os.Stat(ctx.kubeParam); err != nil {
+				ctx.logger.V(0).Infof("%s is not a valid kubernetes version", ctx.kubeParam)
+				return fmt.Errorf("%s is not a valid kubernetes version", ctx.kubeParam)
+			}
 		}
-		ctx.kubeRoot = kubeRoot
 	}
 
-	// initialize bits
-	builder, err := kube.NewDockerBuilder(ctx.logger, ctx.kubeRoot, ctx.arch)
-	if err != nil {
-		return err
+	if ctx.builder == nil && ctx.kubeParam != "" {
+		if info, err := os.Stat(ctx.kubeParam); err == nil && info.Mode().IsRegular() {
```

Comment:
I think I'd feel safer swapping this with the SemVer check above. Maybe it's rare, but if I'm a user testing out things with different versions and I make a local source directory named `v1.30.0`, it might be surprising why my local source isn't being used.

Or maybe a note somewhere declaring the expected order of precedence. Which reminds me - should we update the docs with this change to note the additional options?

Reply:
@stmcginnis good points, let me revise the PR with the table i had above as well.